### PR TITLE
Make oneDNN ACL default on Neoverse V1 cores

### PR DESCRIPTION
**Pull request by cfRod**
This PR makes oneDNN+ACL default for Arm Neoverse V1 cores

### CODE REVIEW

**Comment by cantonios** [comment]
Marked as draft until we have performance numbers to justify the change

**Comment by gbaned** [comment]
@cfRod This PR is in draft, any update on this? Please. Thank you!

**Comment by cfRod** [comment]
Hi @gbaned I have updated this PR 2 weeks ago..

File: tensorflow/tsl/platform/cpu_info.cc
Comment (penpornk) [inline tensorflow/tsl/platform/cpu_info.cc:441]: Why do we turn `clang-format off` here?

File: tensorflow/tsl/platform/cpu_info.cc
Comment (penpornk) [inline tensorflow/tsl/platform/cpu_info.cc:446]: I don't think we need this line since we just had a `return 0;`.

File: tensorflow/tsl/platform/cpu_info.cc
Comment (penpornk) [inline tensorflow/tsl/platform/cpu_info.cc:373]: Nit:
```suggestion
    // Initialize CPUIDInfo pointer.
```

File: tensorflow/tsl/platform/cpu_info.h
Comment (penpornk) [inline tensorflow/tsl/platform/cpu_info.h:137]: Nit: Let's stick with the camel case enum naming.
```suggestion
enum Aarch64CPU {
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and sorry for the delay!
Apart from `cpu_info.h`s, `cpu_info.cc`, and `port.cc`, are there other files related to this PR? I think other files are just from #60026 and #61110. If you don't make changes to them in this PR, could you please exclude them? #60026 is being merged, and I'll merge #61110 before getting to this PR.

File: tensorflow/core/util/port.cc
Comment (penpornk) [inline tensorflow/core/util/port.cc:1]: Can we put the changes in this file in a separate PR?

**Review by penpornk — CHANGES_REQUESTED** [review]
I'm thinking that we can decouple this PR from #60026 and #61110 by splitting the changes in `port.cc` to a new PR. Then this PR is just preparing the necessary aarch64 CPU info, and can be merged anytime. Only the new `port.cc` PR depends on #60026 and #61110 (since we need all the optimizations to get in before turning on TF-oneDNN by default for V1).

File: tensorflow/tsl/platform/cpu_info.cc
Comment (milpuz01) [reply tensorflow/tsl/platform/cpu_info.cc:379]: Thanks @penpornk for the review. In `cpu_info.cc` we are actually reading value of registers from thefile `/sys/devices/system/cpu/cpu<idx>/regs/identification/midr_el1` as suggested in linked issue, but we need to make this check to make sure that CPUID registers are available before reading them.

Comment (milpuz01) [reply tensorflow/tsl/platform/cpu_info.cc:379]: As per above we will add comment describing why the check is required.

File: tensorflow/tsl/platform/cpu_info.cc
Comment (milpuz01) [reply tensorflow/tsl/platform/cpu_info.cc:431]: Yes, there are technical reference manuals and for V1 values for `implementer_` and `cpunum_` (part number) can be found here: https://developer.arm.com/documentation/101427/0101/Register-descriptions/AArch64-system-registers/MIDR-EL1--Main-ID-Register--EL1. We will add reference to them in the comments. Thanks for pointing out.

File: tensorflow/core/util/port.cc
Comment (cfRod) [reply tensorflow/core/util/port.cc:1]: I am going to decouple this PR and address comments related to cpu_info in the new PR.

File: tensorflow/tsl/platform/cpu_info.cc
Comment (penpornk) [reply tensorflow/tsl/platform/cpu_info.cc]: Thank you for the explanation! :)

File: tensorflow/core/util/port.cc
Comment (penpornk) [reply tensorflow/core/util/port.cc:1]: Sounds good. Thank you! :)

File: tensorflow/tsl/platform/cpu_info.cc
Comment (cfRod) [reply tensorflow/tsl/platform/cpu_info.cc]: Add comment to https://github.com/tensorflow/tensorflow/pull/61110/commits/3ff50a6a46f1350e8b4f310a8fdac65e1d7498d5

File: tensorflow/tsl/platform/cpu_info.cc
Comment (cfRod) [reply tensorflow/tsl/platform/cpu_info.cc]: Fixed in https://github.com/tensorflow/tensorflow/pull/61110/files

File: tensorflow/tsl/platform/cpu_info.cc
Comment (cfRod) [reply tensorflow/tsl/platform/cpu_info.cc]: Fixed in https://github.com/tensorflow/tensorflow/pull/61110/files

File: tensorflow/tsl/platform/cpu_info.cc
Comment (cfRod) [reply tensorflow/tsl/platform/cpu_info.cc]: Fixed in https://github.com/tensorflow/tensorflow/pull/61110/files

File: tensorflow/tsl/platform/cpu_info.cc
Comment (cfRod) [reply tensorflow/tsl/platform/cpu_info.cc]: Fixed in https://github.com/tensorflow/tensorflow/pull/61110/files

File: tensorflow/tsl/platform/cpu_info.cc
Comment (cfRod) [reply tensorflow/tsl/platform/cpu_info.cc]: Done

File: tensorflow/tsl/platform/cpu_info.cc
Comment (cfRod) [reply tensorflow/tsl/platform/cpu_info.cc]: Fixed in https://github.com/tensorflow/tensorflow/pull/61110/files

File: tensorflow/tsl/platform/cpu_info.h
Comment (cfRod) [reply tensorflow/tsl/platform/cpu_info.h]: Fixed in https://github.com/tensorflow/tensorflow/pull/61110/files

File: tensorflow/core/util/port.cc
Comment (cfRod) [reply tensorflow/core/util/port.cc:1]: Done. Decouple changes and added it as a separate commit to PR https://github.com/tensorflow/tensorflow/pull/61110

**Review by penpornk — APPROVED** [review]
Thank you for the changes! port.cc looks good to me. I'll hold off on merging this PR until #61110 is merged.

**Pull request merged**
